### ISSUE TITLE
Sonoff Basic lost connection to Wifi after upgrading from 12.2.0 to 12.3.1

### ISSUE THREAD

**Issue by HectorDRR** [arendst/Tasmota#17500]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
I tried to upgrade my Sonoff Basic to the latest version (12.3.1) from 11.0.0 but after upgrade, my sonoff basic didn't connect to my wifi. I've tryed to do a recover and it shows the AP and I can connect but setting up a manual ip address in my moibile because it didn't get IP from the Sonoff. Connect to 192.168.4.1 and I see the normal menu where it detect my network and my wifi password is correct. I save the configuration but after restart the problem is the same. It didn' connect and the led get flasing forever.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Sonoff Basic
- [x] Tasmota binary firmware version number used: 12.3.1
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: OTA
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
14:53:08.650 MQT: stat/bomba/RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}
14:53:08.871 MQT: stat/bomba/RESULT = {"Module":{"1":"Sonoff Basic"}}
14:53:09.126 MQT: stat/bomba/RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}

```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
14:54:17.075 MQT: stat/bomba/RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":87,"Free":424,"Rules":"on system#Boot do backlog pulsetime1 150; Mem1 0 endon on Time#Minute=0 do mem1 0 endon"}}
14:54:17.299 MQT: stat/bomba/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
14:54:17.550 MQT: stat/bomba/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
14:54:45.026 MQT: stat/bomba/STATUS = {"Status":{"Module":1,"DeviceName":"Bomba","FriendlyName":["Bomba"],"Topic":"bomba","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
14:54:45.065 MQT: stat/bomba/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"sonoffs","OtaUrl":"http://ota.tasmota.com/tasmota/release-12.3.1/tasmota.bin.gz","RestartReason":"Power On","Uptime":"0T00:02:06","StartupUTC":"2022-12-24T14:52:39","Sleep":50,"CfgHolder":4617,"BootCount":13,"BCResetTime":"2020-05-07T18:12:03","SaveCount":39,"SaveAddress":"F9000"}}
14:54:45.099 MQT: stat/bomba/STATUS2 = {"StatusFWR":{"Version":"12.2.0(tasmota)","BuildDateTime":"2022-10-17T08:39:23","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"424/699"}}
14:54:45.123 MQT: stat/bomba/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":2,"LogHost":"192.168.3.8","LogPort":514,"SSId":["Hector",""],"TelePeriod":3600,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5AFF002800000000","00000080","00006000","00004000","00000000"]}}
14:54:45.159 MQT: stat/bomba/STATUS4 = {"StatusMEM":{"ProgramSize":626,"Free":376,"Heap":26,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"1440E0","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000000"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62","Sensors":"1,2,3,4,5,6"}}
14:54:45.202 MQT: stat/bomba/STATUS5 = {"StatusNET":{"Hostname":"bomba","IPAddress":"192.168.3.10","Gateway":"192.168.3.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.3.6","DNSServer2":"0.0.0.0","Mac":"2C:3A:E8:07:C9:CB","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
14:54:45.228 MQT: stat/bomba/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.3.12","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_07C9CB","MqttUser":"DVES_USER","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
14:54:45.253 MQT: stat/bomba/STATUS7 = {"StatusTIM":{"UTC":"2022-12-24T14:54:45","Local":"2022-12-24T14:54:45","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":99,"Sunrise":"07:42","Sunset":"15:56"}}
14:54:45.275 MQT: stat/bomba/STATUS10 = {"StatusSNS":{"Time":"2022-12-24T14:54:45"}}
14:54:45.290 MQT: stat/bomba/STATUS11 = {"StatusSTS":{"Time":"2022-12-24T14:54:45","Uptime":"0T00:02:06","UptimeSec":126,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"Hector","BSSId":"98:97:D1:A6:A3:6E","Channel":11,"Mode":"11n","RSSI":22,"Signal":-89,"LinkCount":1,"Downtime":"0T00:00:06"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_
Upgrade to 12.3.1. Upgrading to 12.2.0 gives no problem.

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
Upgrading ato 12.3.1 must work normally but after do it, the sonoff can't connect to wifi.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._
I've tested one by one from 11.0.0 until 12.3.1. All versions works ok except 12.3.1

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by arendst** [comment]
More users seem to have problems with "older" Sonoff devices. I noticed too that after upgrading to v12.3.1 sometimes makes wifi connections difficult and/or ota downloads fail with timeouts.

I've done some testing by reverting changes introduced in v12.3.1 from v12.2.0 but I still haven't found the culprit. It sometimes works correctly for hours, then it stops working for a while.

I noticed this on Sonoff Basic and T1 devices.

**Comment by HectorDRR** [comment]
While I'm reading your comment, I have the sonoff in front of me connected with the serial programmer, because I had to flash by serial to be abe to recover, and it has been working correct from 15 minutes ago and now I've seen how it disconnect from Wifi and get led flashing and get disconnected and after several seconds, it get connected again
14:54:45.290 MQT: stat/bomba/STATUS11 = {"StatusSTS":{"Time":"2022-12-24T14:54:45","Uptime":"0T00:02:06","UptimeSec":126,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"Hector","BSSId":"98:97:D1:A6:A3:6E","Channel":11,"Mode":"11n","RSSI":22,"Signal":-89,"LinkCount":1,"Downtime":"0T00:00:06"}}}
15:02:39.639 APP: Serial logging disabled
15:07:26.855 WIF: Connecting to AP1 Hector Channel 11 BSSId 98:97:D1:A6:A3:6E in mode 11n as bomba...
15:07:36.801 WIF: Connect failed as AP cannot be reached
15:07:37.140 WIF: Connecting to AP1 Hector in mode 11n as bomba...
15:07:47.020 WIF: Connect failed as AP cannot be reached
15:07:48.021 WIF: Connect failed as AP cannot be reached
15:07:48.360 WIF: Connecting to AP1 Hector in mode 11n as bomba...
15:07:59.004 WIF: Connected
15:07:59.255 MQT: Attempting connection...
15:07:59.466 MQT: Connect failed to 192.168.3.12:1883, rc -2. Retry in 10 sec
15:08:10.540 MQT: Attempting connection...
15:08:10.560 MQT: Connected
15:08:10.564 MQT: tele/bomba/LWT = Online (retained)
15:08:10.567 MQT: cmnd/bomba/POWER = 
15:08:10.794 MQT: stat/bomba/RESULT = {"POWER":"OFF"}
15:08:10.797 MQT: stat/bomba/POWER = OFF
I'll try latest tasmota-lite-bin as recommended

**Comment by HectorDRR** [comment]
Tasmota-lite-bin 12.3.1 upgraded without problem and connected to the Wifi.

00:00:00.001 HDW: ESP8266EX
00:00:00.047 CFG: Loaded from flash at FB, Count 42
00:00:00.211 QPC: Reset
00:00:00.224 Project tasmota - Bomba Version 12.3.1(lite)-2_7_4_9(2022-12-16T10:28:11)
00:00:00.863 WIF: Connecting to AP1 Hector Channel 11 BSSId 98:97:D1:A6:A3:6E in mode 11n as bomba...
00:00:05.504 WIF: Connected
00:00:05.755 HTP: Web server active on bomba with IP address 192.168.3.10
15:13:26.352 MQT: Attempting connection...
15:13:26.384 MQT: Connected
15:13:26.388 MQT: tele/bomba/LWT = Online (retained)
15:13:26.389 MQT: cmnd/bomba/POWER = 
15:13:26.394 MQT: tele/bomba/INFO1 = {"Info1":{"Module":"Sonoff Basic","Version":"12.3.1(lite)","FallbackTopic":"cmnd/DVES_07C9CB_fb/","GroupTopic":"cmnd/sonoffs/"}}
15:13:26.408 MQT: tele/bomba/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"bomba","IPAddress":"192.168.3.10"}}
15:13:26.416 MQT: tele/bomba/INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":14}}
15:13:26.424 MQT: stat/bomba/RESULT = {"POWER":"OFF"}
15:13:26.426 MQT: stat/bomba/POWER = OFF
15:13:26.440 RUL: SYSTEM#BOOT performs "backlog pulsetime1 150; Mem1 0"
15:13:26.481 MQT: stat/bomba/RESULT = {"PulseTime1":{"Set":150,"Remaining":150}}
15:13:27.097 MQT: stat/bomba/RESULT = {"Mem1":"0"}
15:13:31.287 MQT: tele/bomba/STATE = {"Time":"2022-12-24T15:13:31","Uptime":"0T00:00:13","UptimeSec":13,"Vcc":3.514,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"Hector","BSSId":"98:97:D1:A6:A3:6E","Channel":11,"Mode":"11n","RSSI":12,"Signal":-94,"LinkCount":1,"Downtime":"0T00:00:07"}}
15:14:12.774 MQT: Attempting connection...
15:14:12.796 MQT: Connected
15:14:12.799 MQT: tele/bomba/LWT = Online (retained)
15:14:12.801 MQT: cmnd/bomba/POWER = 
15:14:16.907 MQT: stat/bomba/RESULT = {"POWER":"OFF"}
15:14:16.909 MQT: stat/bomba/POWER = OFF

**Comment by HectorDRR** [comment]
I've disconnected power and connect again and it has been a while trying to reconnect until it has worked. The Sonoff has not been moved from his place in front of me and at 5 m from the router.
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at FA, Count 44
00:00:00.056 QPC: Count 1
00:00:00.066 Project tasmota - Bomba Version 12.3.1(lite)-2_7_4_9(2022-12-16T10:28:11)
00:00:00.556 WIF: Connecting to AP1 Hector Channel 11 BSSId 98:97:D1:A6:A3:6E in mode 11n as bomba...
00:00:06.480 QPC: Reset
00:00:12.764 WIF: Connect failed
00:00:13.102 WIF: Connecting to AP1 Hector in mode 11n as bomba...
00:00:23.033 WIF: Connect failed as AP cannot be reached
00:00:24.033 WIF: Connect failed as AP cannot be reached
00:00:24.370 WIF: Connecting to AP1 Hector in mode 11n as bomba...
00:00:34.262 WIF: Connect failed as AP cannot be reached
00:00:35.262 WIF: Connect failed as AP cannot be reached
00:00:35.599 WIF: Connecting to AP1 Hector in mode 11n as bomba...
00:00:45.532 WIF: Connect failed as AP cannot be reached
00:00:46.534 WIF: Connect failed as AP cannot be reached
00:00:46.871 WIF: Connecting to AP1 Hector in mode 11n as bomba...
00:00:59.013 WIF: Connected
00:00:59.264 HTP: Web server active on bomba with IP address 192.168.3.10
00:01:00.274 MQT: Attempting connection...
00:01:00.305 MQT: Connected
00:01:00.309 MQT: tele/bomba/LWT = Online (retained)
00:01:00.310 MQT: cmnd/bomba/POWER = 
00:01:00.315 MQT: tele/bomba/INFO1 = {"Info1":{"Module":"Sonoff Basic","Version":"12.3.1(lite)","FallbackTopic":"cmnd/DVES_07C9CB_fb/","GroupTopic":"cmnd/sonoffs/"}}
00:01:00.330 MQT: tele/bomba/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"bomba","IPAddress":"192.168.3.10"}}
00:01:00.337 MQT: tele/bomba/INFO3 = {"Info3":{"RestartReason":"Power On","BootCount":16}}
00:01:00.345 MQT: stat/bomba/RESULT = {"POWER":"OFF"}
00:01:00.346 MQT: stat/bomba/POWER = OFF
00:01:00.360 RUL: SYSTEM#BOOT performs "backlog pulsetime1 150; Mem1 0"
00:01:00.402 MQT: stat/bomba/RESULT = {"PulseTime1":{"Set":150,"Remaining":150}}
00:01:00.604 MQT: stat/bomba/RESULT = {"Mem1":"0"}
00:01:04.784 MQT: tele/bomba/STATE = {"Time":"1970-01-01T00:01:04","Uptime":"0T00:01:07","UptimeSec":67,"Vcc":3.495,"Heap":28,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"Hector","BSSId":"98:97:D1:A6:A3:6E","Channel":11,"Mode":"11n","RSSI":12,"Signal":-94,"LinkCount":1,"Downtime":"0T00:01:01"}}
15:23:17.536 MQT: Attempting connection...
15:23:17.566 MQT: Connected
15:23:17.570 MQT: tele/bomba/LWT = Online (retained)
15:23:17.572 MQT: cmnd/bomba/POWER = 
15:23:17.791 MQT: stat/bomba/RESULT = {"POWER":"OFF"}
15:23:17.794 MQT: stat/bomba/POWER = OFF
15:23:22.424 MQT: stat/bomba/RESULT = {"POWER":"OFF"}
15:23:22.426 MQT: stat/bomba/POWER = OFF
15:24:03.521 MQT: Attempting connection...
15:24:03.540 MQT: Connected
15:24:03.543 MQT: tele/bomba/LWT = Online (retained)
15:24:03.546 MQT: cmnd/bomba/POWER = 
15:24:09.845 MQT: stat/bomba/RESULT = {"POWER":"OFF"}
15:24:09.847 MQT: stat/bomba/POWER = OFF

I'll put it back at this original place, controlling a pump for recirculate hot water before we get a shower, and I'll try to monitorize if there is more wifi problems.

I'll let my others Sonoff Dual and Sonoff TH with version 12.2.0 for a while until confirm the problem with this firm is solved.

Thanks for your support and greetings for your great work with Tasmota!

Happy Christmas!

**Comment by grischard** [comment]
I'm having a similar issue with tasmota nodes not grabbing an ip address from dhcp. They get the subnet, etc., but don't grab the ip address.

The logs when I first try to connect are a bit different - endless loops of 'Connect failed':

```
18:18:32.692 -> 00:03:20.856 WIF: Connecting to AP1 my-AP1 in mode 11n as tasmota-D071E8-4584...
18:18:44.974 -> 00:03:32.523 WIF: Connect failed with AP timeout
```

As a workaround, I can 'force' them to connect by setting `ipaddress1 192.168.1.x` manually:


```
18:22:22.539 -> 00:07:00.132 CMD: ipaddress
18:22:22.539 -> 00:07:00.138 RSL: RESULT = {"IPAddress1":"0.0.0.0 (0.0.0.0)","IPAddress2":"192.168.1.1","IPAddress3":"255.255.255.0","IPAddress4":"192.168.1.1","IPAddress5":"0.0.0.0"}

…

18:22:32.159 -> 00:07:09.144 CMD: ipaddress1 192.168.1.49
18:22:32.159 -> 00:07:09.149 RSL: RESULT = {"IPAddress1":"192.168.1.49 (0.0.0.0)"}
18:23:13.960 -> 00:00:04.012 CMD: restart 1
18:23:13.960 -> 00:00:04.015 RSL: RESULT = {"Restart":"Restarting"}

…

18:23:16.988 -> 00:00:00.548 WIF: Connecting to AP1 GUACAMOLE.BE in mode 11n as tasmota-D071E8-4584...
18:23:19.812 -> 00:00:03.279 CMD: ipaddress
18:23:19.812 -> 00:00:03.284 RSL: RESULT = {"IPAddress1":"192.168.1.49 (192.168.1.49)","IPAddress2":"192.168.1.1","IPAddress3":"255.255.255.0","IPAddress4":"192.168.1.1","IPAddress5":"0.0.0.0"}
18:23:21.221 -> 00:00:04.503 WIF: Connected
```

My dhcp server is dnsmasq 2.87 running on openwrt. A packet capture there shows that the dhcp server receives a dhcp discover from tasmota, sends a dhcp offer, but never hears back with a dhcp request.

I'm attaching a packet capture:

[tasmota-dhcp.dmp.gz](https://github.com/arendst/Tasmota/files/10300776/tasmota-dhcp.dmp.gz)

**Comment by Extreme79** [comment]
Same issue here. I upgraded to 12.3.1 a sonoff th16. No more connection to the wifi.
After a reset I was able to connect to the tasmota recover ssid. 
I uploaded the config backup and I lost connection again. Reset. Tasmota AP connection. Manual Wifi setup.. connection lost again.
Notice: the sonoff is able to connect to my phone in wifi tethering mode near the sonoff, with the same ssid and pwd I have in my home.. very strange.
After another reset, after a firmware downgrade, uploading the same config backup.. bring the device up n the network again.

**Comment by stefanbode** [comment]
I have this issue that some devices lost connection and do no reconnect already in 12.1. My first guess was an issue with the new Fritzbox 7490AX router but this is unlikely because the repeaters did not change. I also do not have the problem on all devices. Some work great some not. i changed some to wifimode 2, but cannot see that this really makes a difference. anyhow all of them connect correctly AFTER a power off cycle

**Comment by arendst** [comment]
@HectorDRR I see this issue is already being hijacked by others so let's continue by calling you directly.

I still cannot stable reproduce the network issue from v12.3.1 on sonoff. What I did is adding debug ``SetOption177`` to disable a feature introduced in v12.3.1. Pls try to load latest dev release and enable ``SO177 1``. This disables a feature running during the sleep period which might make the Sonoffs using too much power resulting in wifi stability errors.

**Comment by Bournetoride** [comment]
"This disables a feature running during the sleep period which might make the Sonoffs using too much power resulting in wifi stability errors."
I noticed something similar on my sonoff basic running 12.0.2.1 , turning off dynamic sleep worked for me

**Comment by grischard** [comment]
@arendst sorry for the hijack! Should I open a separate issue?

**Comment by stefanbode** [comment]
Let me check on the dynamic sleep. Just for test

**Comment by bhp20** [comment]
I had the same problem (upgrading to 12.3.1 OTA) on an old Sonoff Basic R2 V1.0. (BTW, two other same vintage Sonoff's upgraded with no problem).
I had to remove the Sonoff from the wall (of course, the two Sonoffs with no problem were external!) to reflash (using an FTD1232 and Tasmotizer).  This worked.
However, Tasmotizer only allows .bin, and I've been using .bin.gz files for a while; I didn't notice that the last .bin I had was ver 11, until after I squeezed everything back into the wall. When I upgraded (OTA) to the latest, it failed again.
This time I reflashed (via hardwire FTD1232) with the newest firmware (12.3.1.2) which is working.
I am concerned about upgrading OTA next time (I never want to pull the module out of the wall again).
The Sonoff that didn't upgrade has a different configuration than the two that upgraded - it has a capacitive touch switch.
Hope that this helps.

**Comment by mikep1998** [comment]
FYI I'm have the same problem with a D1mini just upgraded to 12.3.1.2 and now can not connect.

**Comment by mikep1998** [comment]
After a couple of hours it was able to connect.

**Comment by stefanbode** [comment]
Following changes: 12.3.1.2 got offline after some days. Reboot and setoption60 1. No problems until now. Now changed back setoption60 0 and setoption177 1. Will see if the device stay online

**Comment by stefanbode** [comment]
ok the device got OFFLINE because of weak network  but recovers without problems after the network was in reach again. Maybe this helps for debugging. setoption 177 == 1

**Comment by stefanbode** [comment]
![image](https://user-images.githubusercontent.com/16225290/210315164-36d9a201-41a7-4155-9bd7-9d588a9cb686.png)
I do not understand why the downtime is still low. This is definitely wrong. The device was OFFLINE for several hours. Looks for me like the device is not recognizing the problem.

**Comment by stefanbode** [comment]
Device is offline again since 15min. No ping possible anymore.
![image](https://user-images.githubusercontent.com/16225290/210359767-d750e1fc-bec0-4a59-a439-a3f282817777.png)

**Comment by Jason2866** [comment]
Is it a Sonoff Basic R2? If yes, best solution is too ditch in the bin.
Device is a design fail in every aspect. Power supply is way to weak to power the esp reliable.
Next thermal design of esp and flash chip. The esp is mounted to heat the flash.
Have a user example where this damaged the flash. Replaced the flash and device was functional again. BUT esp gets to hot because it is not powered correctly and the esp seems to try to compensate this. So power supply is stressed more. Bad powering is resulting in wifi issues. This is a known dependency of the esp8266.
So if the device connects to wifi without issues when powered via USB serial adapter, the power supply is to blame. Replace or fix the power supply and the device will work.

**Comment by Jason2866** [comment]
For the Sonoff Basic R2 the last working Tasmota version is v12.2.x
Reason: Later version has more features which has more MCU load, and needs more current.
The poor weak power supply of the R2 can’t handle this.

**Comment by Jason2866** [comment]
Please check if you have set `sleep 20`. Sleep is needed for the R2 !!

**Comment by Extreme79** [comment]
> For the Sonoff Basic R2 the last working Tasmota version is v12.2.x Reason: Later version has more features which has more MCU load, and needs more current. The poor weak power supply of the R2 can’t handle this.

Also valid for sonoff TH16 (where I have the same issue)?

**Comment by Jason2866** [comment]
How old is the TH16? All my Sonoff Basic i had in use are dead. The won’t last long.
Sonoff devices are built from cheapest components and hardware design is bleeding edge.
Repaired a few for testing… No Sonoff Basic or sort off anymore in productive use.
There are way better alternatives for the Sonoff Basics.

**Comment by wchulze** [comment]
I've been using 24 Sonoff Basics in all versions for 4 years with Tasmota without a failure or problem. If Tasmota no longer works for Sonoff the "S" in the name Tasmota is no longer valid. A new name would be a good indication of the situation. That's not meant to be taken seriously.
Please provide versions with less software load. Because the original Sonoff software is adapted to the hardware. The problem is therefore not in the hardware but in the Tasmota software that does not fit.

**Comment by Jason2866** [comment]
@wchulze There is a liteversion. Tasmota–lite. We can’t do more.
As shown before latest Tasmota does run on my known very bad flaky R2.
 
The „s“ in Tasmota is just historic. Ideas for reuse of the „s“ in naming are welcome.

**Comment by sfromis** [comment]
As the Sonoff Basic R2 is a worse device than the older Sonoff Basic, which was the starting point of Tasmota, Tasmota was not downgraded to the crappy R2.

**Comment by Jason2866** [comment]
How about. Tasmota. (T)heo (A)rendst (S)mart (M)qtt (OTA) ?

**Comment by wchulze** [comment]
@[Jason2866](https://github.com/Jason2866) Good idea.
The Lite version is only available in English. How can I compile it - locally - into another interface language? Previously, in older versions I could set a compiler switch.

**Comment by barbudor** [comment]
Everything is in https://tasmota.github.io/docs/Compile-your-build/

The easiest way for local compile is installing Jason's Portable VSCode package which include everything

**Comment by Jason2866** [comment]
```
[env:tasmota-lite-de]
build_flags             = ${env.build_flags} -DFIRMWARE_LITE -DMY_LANGUAGE=de_DE
lib_extra_dirs          =
```

**Comment by Extreme79** [comment]
> How old is the TH16? All my Sonoff Basic i had in use are dead. The won’t last long.

Mine was bought 3yrs ago and used to drive a fridge compressor.
  I solved reinstalling a previous firmware, not a real problem.. Newer firmware is not really needed.
  It's more like curiosity about the issue that not a real need.
  Because in my case with the new firmware I got no connection on the router BUT, placing my phone as Hotspot, configured as my router... Bum! Got the sonoff connected to the phone and perfectly usable.

This mean that the issue is not, probably linked to what I can read above.

**Comment by wchulze** [comment]
@[barbudor](https://github.com/barbudor) Thank,s for info, but I know.
@[Jason2866](https://github.com/Jason2866) Got it.  Now the PID controler fails to comiple due to missing timeprop.h now. Maybe I have to open an new issue.

**Comment by Extreme79** [comment]
> @Extreme79 Probably a router DHCP cache issue. Delete cache (easiest restart Router) and it will connect. Not related.

I think not.
I obviously already tested a router reboot and also a reduced dhcp lease time, but nothing works (all my iot devices has a static dhcp assignment by Mac address, but anyway same issue disabling static ip assignment).
The only way to get it back connected, it's to downgrade the firmware

**Comment by Jason2866** [comment]
@Extreme79 okay, but not related to this.

**Comment by Jason2866** [comment]
> @[barbudor](https://github.com/barbudor) Thank,s for info, but I know.
> @[Jason2866](https://github.com/Jason2866) Got it. Now the PID controller fails to compile due to missing timeprop.h now. Maybe I have to open an new issue.

Tasmota–lite can not include specific stuff. It is as it is. Nothing can be added.

**Comment by Jason2866** [comment]
@Extreme79 please open a discussion with many infos. Code for DNS resolving has changed.

**Comment by wunderbaum** [comment]
I have the same issue with my wemos-boards.
As I wanted to experiment with the new LD2410 sensor, I had to compile my own firmware on [benzino77s compiler](https://gitpod.io/#https://github.com/benzino77/tasmocompiler) page.
I only choose 
Alexa, HomeAssistant, mDNS, Web Interface, Rules and Timers.
And I added: #define USE_LD2410 to the user_config_override.
The firmware was only 588k large and I was able to flash it to a wemos-board. But it cannot connect to my WLAN. Every firmware-version lower than 12.3.1 worked perfect. What can I do to get a better result?

**Comment by Jason2866** [comment]
@wunderbaum if you say Wemos you probably mean Wemos D1 mini clones.
The other known trouble makers. Search for weak LDO Wemos D1 mini. You will find many hits.
Solution replace the two weak onboard 3.3V LDO used on nearly all clones, with a 500mA type which is used on the orig. Wemos D1 mini. That's all.

**Comment by wunderbaum** [comment]
@Jason2866 Ah ok. Its just a cheap Wemos (not mini). I took it out of the foil and flashed it.
Then I tried the previous versions and they all worked. The LDO is an AMS1117.

Something severe must have been changed. I understand that some old devices can't be fit new features any more, so I point out that there should be some kind of check that nobody renders his/hers device to death, especially when its behind tons of concrete.

**Comment by Jason2866** [comment]
@wunderbaum It is a sort of a NodeMCU. Should work. Have no device with this behaviour.
Without a device where the behaviour is reproducible it is not possible to find the reason for.
Not saying there is maybe a change which does generate that.
So we can not say anything.

**Comment by Jason2866** [comment]
@wunderbaum can you please test if the latest development `Tasmota.bin` the English one NOT localised one is working. We had one issue in the past where a special character in the language file did broke Tasmota firmware.

**Comment by wunderbaum** [comment]
@Jason2866 I tried this one: http://ota.tasmota.com/tasmota/tasmota.bin (12.3.1.5) on that wemos.
Still the same.
I tried a shelly1, too. That works.
Both where connected via USB-Adapter of my PC, so shelly without mains.

**Comment by Jason2866** [comment]
Do you have an oscilloscope? If yes could you check the 3.3V directly on the esp8266?
Searching for spikes and drops.

**Comment by wunderbaum** [comment]
I have access to one, but not tonight. I try to connect one soon and do some measurements. We have a Rigol 1052 in the lab I hope.

**Comment by stefanbode** [comment]
I used the AMS1117 in many of my projects so far without any problems. If it is a fake one, maybe this can cause the issue. Additional I see two problems: some users are even not able to connect the first time and others like me get connected most time but have a non recoverable lost after some time. But all is 12.3 related as far as I can see. If power consumption is significant higher in 12.3 this might explain the LDO stuff. Most devices deliver enough in 5V but this means nothing if the LDO fail to provide the current. But here a capacitor on the 3.3V rail really should work and fix it. If it is the problem....

**Comment by wunderbaum** [comment]
Good morning,
I oscilloscoped two wemos, one with 12.2.0, and the other one with 12.3.1.5. There was no difference on 3.3V pin. At least for my eye. 
When I do a _reset 1_, I can see the AP for configuration, but I do not get an IP. In 12.2.0, I get one.

**Comment by Jason2866** [comment]
The part where an AP is opened needs the most current. The behaviour with connection issues to wifi or not seeing an device AP is typical for powering issues.
The defect Sonoff R2 which I got from an user had an (defect) XMC flash chip. After replacing it started badly working again. Seeing bad wifi rssi. If powered from extern rssi is perfect. Maybe XMC flash are somehow related to this issue? Would be interesting to replace the XMC flash against a Winbond flash. To see if there is a chance.
Always have not a good feeling seeing HF designs without a metall shield...

**Comment by wunderbaum** [comment]
But what is causing this high demand for current as it worked in all previous versions?
How can I reduce the TX power?

**Comment by Jason2866** [comment]
See [here](https://tasmota.github.io/docs/WifiPower/)
We don't know the reason. The symptoms seen are typical for powering issues. It can be something else. If we would know, what the reason for causing we would try to change

**Comment by wunderbaum** [comment]
Strange, now I compiled version 12.3.0  with TasmoCompiler, and that version still works.
I always use the same device and power supply for testing.

**Comment by Jason2866** [comment]
Do NOT use 12.3.0. it is replaced from 12.3.1 since the 12.3.0 has bug.

**Comment by wunderbaum** [comment]
So, next round in my investigation. 12.3.0 only works with the packages Webinterface, LittleFS, Rules, Timers. Adding one more package makes everything very unstable. 
Yes, TasmoCompiler only includes the choosen packages.

**Comment by Jason2866** [comment]
We have made a version like this. `Tasmota-lite`

**Comment by wchulze** [comment]
[wunderbaum](https://github.com/wunderbaum) When I do a reset 1, I can see the AP for configuration, but I do not get an IP. In 12.2.0, I get one.

[Jason2866](https://github.com/Jason2866) The part where an AP is opened needs the most current. The behaviour with connection issues to wifi or not seeing an device AP is typical for powering issues.

I am amazed at the discussion. The AP is opened. The IP is missing. So this is supposed to be a power supply problem?

**Comment by wunderbaum** [comment]
@wchulze So, at least we are 3 participants that are investigating. Unfortunately far away from swarm intelligence as this problem seems to be very strange.
Until version 12.2.0, I can flash different firmware-versions (sensors, language, etc) on my wemos board (have a second one to crosstest) and always it connects.
@Jason2866 I tried  12.3.1.5(lite) and that connected after some trials according to the serial log.

On a Shelly1v3 the official release 12.3.1 worked. I search my components for older ones for further investigation.

**Comment by Jason2866** [comment]
@wunderbaum You have still better position than we do. Can't reproduce the problem at all.

@wchulze 
> I am amazed at the discussion. The AP is opened. The IP is missing. So this is supposed to be a power supply problem?

Most probably yes. The esp produces "nice" issues when power supply is not stable.

**Comment by wunderbaum** [comment]
@Jason2866 It depends, I would like more to be in a position where this strange problem not appears, too :-)
Today, I tried with a raw ESP12E (ancient model) in a flashing board. That I flashed and was able to update it. Runs without any issue.
It pains me that I can't search much deeper. (In german: It worms me).

So I am a little worried that I will brick a device that is not that easy accessible. Of course, I update only devices where I think its really necessary because it needs one of the new features.

**Comment by wchulze** [comment]
I am afraid this will continue
In the WEMOS clones, which worked well with Tasmota so far, a "4A2D" = XC6204 low drop-out regulator is installed. It can deliver approx. 300 mA in the peak.

You can find it at 
https://tasmota.github.io/docs/Energy-Saving/#quantity-measurement-of-esp-12-moduleesp8266-33v-line
a good basis for the power supply of an ESP-12 and Tasmota.

Please note the spikes with about 320 mA. With this it was just possible to use Tasmota.
At the moment I'm observing that Tasmota should no longer be installed with a clear conscience due to increased requirements on this hardware - a pity. In my opinion, this applies to many other devices that have worked without any problems so far. I hope it doesn't go on like this.

**Comment by Jason2866** [comment]
@wchulze SMD marking code 4A2D is Torex XC6204. From this LDO there are 150mA and 300 mA versions. The 300mA is the XC6204 E to H type. The SMD marking code 4A2D gives no clue on a 150 or 300mA version.

**Comment by jpduhen** [comment]
Sonoff basic has a 

> Is it a Sonoff Basic R2? If yes, best solution is too ditch in the bin.
> Device is a design fail in every aspect. Power supply is way to weak to power the esp reliable.
> Next thermal design of esp and flash chip. The esp is mounted to heat the flash.
> Have a user example where this damaged the flash. Replaced the flash and device was functional again. BUT esp gets to hot because it is not powered correctly and the esp seems to try to compensate this. So power supply is stressed more. Bad powering is resulting in wifi issues. This is a known dependency of the esp8266.
> So if the device connects to wifi without issues when powered via USB serial adapter, the power supply is to blame. Replace or fix the power supply and the device will work.

This helped for me, perhaps also useful for others: 
[https://youtu.be/H74POOYlqsU](url)

**Comment by wchulze** [comment]
[Jason2866](https://github.com/Jason2866)Vin MUST be at least 2V greater than Vout (2V dropout). USB is 5V. So this choice is always bad.
No, the data sheet specifies max 330 mV
With a 2 volt dropout, the target value of 3.3 V would not even be reached

**Comment by wchulze** [comment]
@jpduhen Thanks for the link to the interesting video. The charging capacitors for switching power supplies are critical because high alternating currents flow through them (large capacity and high frequency). Therefore, they can age quite quickly. When replacing the electrolytic capacitor, care should be taken to ensure a low series resistance and series inductance (low esr type). A type so simple from the craft box is usually bad. The additional 100nF reduced the existing series inductance.

**Comment by Jason2866** [comment]
@wchulze yep, did look wrong. 200mV. But the problem persist you don't know which current the one is soldered can deliver. NONE of my Wemos D1 mini with replaced LDO have an issue with **any** Tasmota version. Since you have a problematic one, do the replacement of the LDO and let us know your result. The video for the Sonoff Basic shows it is the power supply too.
Tasmota will not drop features and try by doing this to support faulty hardware designs.

The currently one and only case which is unclear why it fails is the one from @wunderbaum

**Comment by wchulze** [comment]
I solved the recognition of the LDO as follows:
Connect the Wemos to a 5V laboratory power supply
Press and hold the Wemos reset button
Carefully load t the 3 volt output with an electronic load (possibly with a resistor)
View the current flow (feeding characteristic)

**Comment by stefanbode** [comment]
I also watched the video. And in my view the original capacitor is not faulty. It just cannot deliver short term current. Adding the 104 in parallel is definitely something I will give a try because very easy to realize. And it also make sense from the electric point.

**Comment by Jason2866** [comment]
> I am afraid this will continue In the WEMOS clones, which worked well with Tasmota so far, a "4A2D" = XC6204 low drop-out regulator is installed. It can deliver approx. 300 mA in the peak.
> 
> You can find it at https://tasmota.github.io/docs/Energy-Saving/#quantity-measurement-of-esp-12-moduleesp8266-33v-line a good basis for the power supply of an ESP-12 and Tasmota.
> 
> Please note the spikes with about 320 mA. With this it was just possible to use Tasmota. At the moment I'm observing that Tasmota should no longer be installed with a clear conscience due to increased requirements on this hardware - a pity. In my opinion, this applies to many other devices that have worked without any problems so far. I hope it doesn't go on like this.

So why talking all about, when no Issues? As said orig. Wemos Mini D1 have a 500mA LDO...

**Comment by ChrisThomas42** [comment]
I'm not sure if this helps or the correct place to post but...  SInce updating from 12.2.0 to 12.3.1, one of my Sonoff Mini R2s disappears off the network frequently.  I could see it was connecting to my router rather than a nearby AP and I thought this was the issue.  I got it to connect to the AP, but it reverted to the router sometime during the day.

I wasn't having any issue with any device disconnecting/becoming unavailable prior to updating and the device was reporting RSSI around 80 when I looked, it's now reporting RSSI between 20 and 40, I've not seen it go above 40 since the upgrade.

I was about to move it again when I came across this discussion, can I help by getting some info before I do anything else?  What info would help? 

I think I'll downgrade my Sonoffs to 12.2.0 otherwise?

Chris

**Comment by ChrisThomas42** [comment]
@Jason2866 Yes, I understand that.  I was only reporting what I'm seeing at the device in that more data points might give a clue to what is happening.  Prior to updating to 12.3.1 I was not having any connectivity issues.

Before I downgrade (my Sonoffs) to 12.2.0 is there any info I can gather to help?

**Comment by Jason2866** [comment]
@ChrisThomas42 Try `tasmota-lite`

**Comment by ChrisThomas42** [comment]
So... I DO have a problem with connection to my Sonoff Minis. I've done some testing and I get the same problem with 12.2.0 and 12.3.1 full fat and lite versions, so I've eliminated the changing Tasmota versions.  More testing/investigations and thanks to @Jason2866.

**Comment by Jason2866** [comment]
@ChrisThomas42 Do you have an Asus Router/AP? Or an other bleeding edge full featured device?

**Comment by ChrisThomas42** [comment]
@Jason2866  Ubiquiti Dream Router, with a couple of APs.  Installed almost a year ago not noticed any issues until I updated Tasmota, that is I hadn't noticed if there were any issues.  I can be 100% sure there wasn't an issue but all my automations worked, then I noticed that one of the automations hadn't worked.  Investigated and noticed the Sonoff mini was dropping off the network regularly.  Since I've been monitoring, the other Tasmota devices have no issues, whether nearer or further from Router or AP.

**Comment by Jason2866** [comment]
Not the WLAN if other esp8266 with Tasmota do work fine. So the Sonoff Minis seems to have a weak power supply too or/and a bad antenna design.
Disconnect one from Mains and inject 5V DC where normally the 5V output from the onboard power supply is. If the device is working well, you know it is the power supply.

**Comment by ChrisThomas42** [comment]
@Jason2866 I have two Sonoff Minis. Both display a lower RSSI than a nearby non Sonoff Tasmota device.  I also have two Sonoff MiniLs that are Zigbee and are not showing any connectivity issues.

I need to have a think what the best solution for me would be.  I will, at some point, investigate the 5V supply on the Minis.  I appreciate your help!

**Comment by Jason2866** [comment]
@ChrisThomas42 a low RSSI is a symptom of bad powering the esp8266, if a similar device at the same place does have a way better RSSI.

**Comment by stefanbode** [comment]
I just added a 104 capacitor one of the devices as shown in the video. Let's have a look if this stabilize

**Comment by gfn256** [comment]
@stefanbode @I have followed this whole discussion/issue and I'm very interested myself, since I have a TH16 that has similar problems. I also believe its a power problem, since by removing the temp/humd sensor from jack, its seems to be much more stable with wifi connection.
I am interested in adding a 104 cap and would really like to know the feedback you got after adding the cap.
**On the TH16 do you know the best place to add it.** I noticed in the video posted, that he did not add it in direct parallel to the bigger cap. Any idea why?
I would also like to note, that I have a pretty long cable to the sensor (2-3m), and maybe I need to add another cap somewhere on that line. Any ideas welcome.

**Comment by wchulze** [comment]
So far, a couple of cases have been reported in this thread where problems arose after an update from Tasmota 12.3.0 to 12.3.2. There is a suspicion that it is a power supply problem. Funny, if downgrade the error disappears reproducible. To get to the reason of the matter, I measured the power consumption with an oscillgraph. The measurement setup looks like this

![image](https://user-images.githubusercontent.com/41591541/218305478-4af87cba-2d98-461f-9dbc-1f333d5c74df.png)

### Tasmota 11.1.0

![image](https://user-images.githubusercontent.com/41591541/218305549-466e2647-479c-48bb-b526-8e47ae3847ee.png)

The current peaks have a maximum length of approx. 400 µs and do not exceed the 200 mA value

![image](https://user-images.githubusercontent.com/41591541/218305561-8854a2a2-f6cd-4922-ba9a-8c44564e518f.png)


### Tasmota 12.3.1

![image](https://user-images.githubusercontent.com/41591541/218305614-530683d9-1222-4b46-8b64-32a387b1adb2.png)

The current peaks have a maximum length of approx. 400 µs and do not exceed the 200 mA value

![image](https://user-images.githubusercontent.com/41591541/218305627-cb48ef74-9b22-451f-ad70-ca533780bb1c.png)

Both versions show similar behaviour.
There are no significant changes recognizable that indicate problems with the power supply when upgrading to 12.3.1.
This suggests to me that there is a change in the Tasmota firmware.

**Comment by wchulze** [comment]
![image](https://user-images.githubusercontent.com/41591541/218306659-2a42c92d-f988-4da6-8d95-4ee6d153bc60.png)

**Comment by Jason2866** [comment]
Looking at, there are more current draw spikes with 400us in v12.3.1 than in 11.1.0.
The overall current draw is higher.

**Comment by wchulze** [comment]
@[Jason2866](https://github.com/Jason2866) Looking at, there are more current draw spikes with 400us in v12.3.1 than in 11.1.0.

No, on average the power consumption is identical. Why should more features increase power consumption? This can only lead to a slightly longer time in the schedule() part of Tasmota. During this time, the system only takes about 80 mA. However, every power supply should be able to do this.

**Comment by Jason2866** [comment]
Seeing more spikes in Screenshot for v12.3.1. More current needed. More feature less idle time in Tasmota sleep routine. Same as for every other CPU more load more heat (=current)

And no, not every power supply can handle. That's why we did the Tasmota lite Version and implemented the Tasmota sleep routine. Now this saving is not enough anymore to get devices with real bad power supplys going.
You are ignoring the fact that changes (adding caps) does solve the issue.

**Comment by wchulze** [comment]
@Jason286 _You are ignoring the fact that changes (adding caps) does solve the issue.

Where does this knowledge come from? In the video, an electrolytic capacitor, perhaps defective, against a possibly unsuitable electrolytic capacitor (bad esr and inductivity), a 100 nF capacitor solves the problem, wow

**Simplified**
I*t=C*U
I = current, T = time, C = capacitance, U = voltage
Assume delta U = 100 mV, I = 100 mA, C = 100 nF
t = C * U / i = 0.1 µs
Good against high-frequency interference

**Comment by Jason2866** [comment]
You did the assumption or better guess

> However, every power supply should be able to do this.

You have not shown any data to verify this.

I do have a Sonoff Basic R2 where all caps are replaced and a few 100nF caps added too.
This device works fine with every Tasmota version. Got this device non working from a user with Tasmota a few years ago. The problem with the R2 is an old story. Andre Thomas and i  did investigate and found the reason. Bad Hardware. Andre developed the Tasmota sleep routine, so we have reduce the needed current draw and provided the Tasmota lite Version. Explained already in this thread...

**Comment by gfn256** [comment]
@Jason2866 On the TH16 could you show me where to add caps, as this would help me immensely. Thanks.

**Comment by Jason2866** [comment]
@gfn256 Dont have or had a TH16. So i don't know the board design. All my Sonoff Basics i had died after 2 years. Desoldered the esp8266 from. Everything else has gone to the bin tray. No Sonoff Basic left in my house. Replaced with more reliable alternatives. Just kept the repaired R2 for tests. The one Sonoff Pow i bought is still working. Had more luck with or the design is better.

**Comment by wchulze** [comment]
@Jason2866 You have not shown any data to verify this.

The clock frequency of an SMPS is typically 100 kHz (100 ns / clock cycle). 400 µs current pulses can be seen in the digrams. For the pulse this are 4000 clock cycles to deliver 200 mA, an almost eternal time. If the SMPS could not supply this current, it would shut down. Why should such a power supply not be able to deliver 80 mA? That it can is to be recognized in the pictures (200 msec).

**Comment by Jason2866** [comment]
> Why should such a power supply not be able to deliver 80 mA? That it can is to be recognized in the pictures (200 msec).

Verified not working in real life. Get such a problem device and troubleshoot. Your fix will be done in the power supply.

**Comment by wchulze** [comment]
Fruitless dicusion

**Comment by wchulze** [comment]
A question for the community:
Has anyone noticed or investigated if the problem of WiFi changes when the relay is switched on or off within the board? The relay needs about 70 mA current.  If you are worried about the power supply, you may  find a difference.

**Comment by sfromis** [comment]
As the relay is not powered from the 3.3V supply, relay engagement may not be much of a predictor for wifi issues due to the bad hardware having power delivery issues.

**Comment by wchulze** [comment]
The Sonff Basic Power supply ( public )

![image](https://user-images.githubusercontent.com/41591541/218326876-6a60c866-23cd-4d21-b36f-8b5761e2435f.png)

As can be seen, the 3.3V rail is supplied by the 5V. If the LDO is ok, the 3.3 V voltage remains stable until the 5V is out of range. We have a pretty unique dependency.

**Comment by gfn256** [comment]
@wchulze I have looked to see (on TH16) a difference with relay on off, and could not find any correlation to wifi access. However (as mentioned above) a couple of days ago, I removed the (AM2301?) temp/humd sensor from the jack, and also the Si7021 setting GPIO4 setting in the config, and since then its seems much more stable.
What bothers me the most in this, is that prior to flashing the TH16 with Tasmota, I never had any access issues with the native Ewlink app. (Though, there is no chance of me going back there!).

**Comment by wchulze** [comment]
@[gfn256](https://github.com/gfn256) Thanks for the info.
The thing about the sensors could be radiation. Try placing a small ferrite bead over the sensor leads. I'm not sure.

**Comment by gfn256** [comment]
@wchulze 1. Hard to believe as it worked flawlessly with original Sonoff firmware.  2. I don't have any ferrite beads to test with!

**Comment by wchulze** [comment]
Yes, that is quite saddening

**Comment by sfromis** [comment]
I've seen no Wifi connectivity issues with any of my 3 Sonoff TH16 devices, while repeatedly upgrading through current Tasmota versions up to now latest dev binary. No signs of any inherent Tasmota issues with these devices, but I don't know if there could be differences with another hardware revision.

Only problem I've had was a SI7021 sensor probe no longer working, but as this issue followed the probe when moving it to another TH16, that very much looks like the probe became defective.

**Comment by wchulze** [comment]
@sfromis, gfn256
Please try to find out together where the differences in your TH16 devices could be. Such as versions, configurations, rules. RSS. field strength ....
It is not very helpful to simply assume differences in the hardware.

**Comment by delinend** [comment]
Also many problems with Tasmota, after upgrade from 12.2.x to 12.3.x

Sorry to Jailebreak this thread..

All my H801 LED modul (modul 18) (20 pices) are rebooting randomly, from 2-20 minutes intervals. They are blinking/flasning LED strip's,  when rebooting. Reboot reason is Hardware_Watchdog.

They are offen rebooting, on a MQTT light set from Domiticz, or on MQTT 5 minutes update interval, from device/Tasmota to MQTT server. Generaly on all some sort og WiFi trafic.

All H801 moduls, have there own powersupply's (12V - 4A, 6A and 10A). Cabeling length from powersupply to H801 module, ~10cm. So they are not missing any powers.

Downgrade to 12.2.x and the problem is gone.

It only happens on 100% PWM load, with one or more LED channels. If running on ex. 30% PWM load, there is no problems with rebooting.

Regaring SONOFF Basic's.. I have 10 pices running. 5 old with esp8266 and 5 new R2 with ESP8285. They both (versions) behave stange in AP mode with Tasmota 12.3.x.  Also when high MQTT trafic.

First I was replacing Flash chip's, but after I tryed brand new moduls from my stock, I saw that the problem was the same, on all devices.

Here some log's, from my H801's, after reboot (Hardware_Watchdog). 100% load on 4 channels:

> 00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F8, Count 20540
00:00:00.055 QPC: Count 1
00:00:00.083 Project tasmota - H801_B94A84 Version 12.3.1.6(tasmota)-2_7_4_9(2023-02-09T15:03:47)
00:00:00.503 WIF: Connecting to AP1 dims Channel 6 BSSId BE:FB:E4:44:09:D2 in mode 11n as H801-B94A84...
00:00:02.758 WIF: Connected
00:00:03.006 HTP: Web server active on H801-B94A84 with IP address 192.168.1.28
06:41:01.002 MQT: stat/H801_B94A84/RESULT = {"GPIOs2":{"6656":"TM1637 DIO","6688":"DLP Tx","6720":"DLP Rx","6752":"SSD1351 DC","6784":"XPT2046 CS","6816":"CSE7761 Tx","6848":"CSE7761 Rx","6880":"VL53LXX XSHUT","6912":"MAX7219 CLK","6944":"MAX7219 DIN","6976":"MAX7219 CS","7008":"TFmini+ TX","7040":"TFmini+ RX","7072":"ZC Pulse","7104":"Input","7136":"I2S DOUT","7168":"I2S BCLK","7200":"I2S WS","7232":"I2S DIN","7264":"I2S WS IN","7296":"I2S BCLK IN","7328":"Interrupt","7360":"MCP2515 CS","7392":"HRG15 Tx","7424":"HRG15 Rx","7456":"VINDRIKTNING","7488":"BL0939 Rx","7520":"BL0942 Rx","7552":"HM330X SET","7584":"Heartbeat","7616":"Heartbeat_i","7648":"74x595 SRCLK","7680":"74x595 RCLK","7712":"74x595 OE","7744":"74x595 SER","7776":"SolaxX1 RTS","7808":"Option E","7840":"SDM230 Tx","7872":"SDM230 Rx","7904":"ADC MQ","7936":"CM110x TX","7968":"CM110x RX","8000":"BL6523 Tx","8032":"BL6523 Rx","8064":"ADE7880 IRQ","8096":"Reset","8128":"MS01","8160":"SDIO CMD","8192":"SDIO CLK","8224":"SDIO D0","8256":"SDIO D1","8288":"SDIO D2","8320":"SDIO D3","8352":"Flowrate","8384":"BP5758D Clk","8416":"BP5758D Dat","8448":"SM2335 Clk","8480":"SM2335 Dat","8512":"MP3 Busy","8544":"TM1621 CS","8576":"TM1621 WR","8608":"TM1621 RD","8640":"TM1621 DAT","8672":"Relay_b","8704":"Relay_bi","8736":"I2S MCLK","8768":"ModBr Tx","8800":"ModBr Rx","8832":"ADE7953 RST","8864":"NrgMbs Tx","8896":"NrgMbs Rx","8928":"ADE7953 CS","8960":"Dali RX","8992":"Dali TX","9024":"BP1658CJ Clk","9056":"BP1658CJ Dat","9088":"Dingtian CLK","9120":"Dingtian SDI","9152":"Dingtian Q7","9184":"Dingtian PL","9216":"Dingtian RCK","9248":"LD2410 Tx","9280":"LD2410 Rx","9312":"ModBr Tx Ena","9344":"NrgMbs Tx Ena","9376":"ME007 Tri","9408":"ME007 Rx","9440":"TuyaMCUBr Tx","9472":"TuyaMCUBr Rx","9504":"BioPDU PZEM0XX Tx","9536":"BioPDU PZEM016 Rx","9568":"BioPDU Bit"}}
06:41:01.165 MQT: stat/H801_B94A84/RESULT = {"SwitchDebounce":50}
06:41:01.217 MQT: stat/H801_B94A84/RESULT = {"BlinkTime":10}
06:41:01.266 MQT: stat/H801_B94A84/RESULT = {"MqttLog":0}
06:41:01.318 MQT: stat/H801_B94A84/RESULT = {"PulseTime2":{"Set":0,"Remaining":0}}
06:41:01.368 MQT: stat/H801_B94A84/RESULT = {"PulseTime4":{"Set":0,"Remaining":0}}
06:41:01.420 MQT: stat/H801_B94A84/RESULT = {"PulseTime6":{"Set":0,"Remaining":0}}
06:41:01.484 MQT: stat/H801_B94A84/RESULT = {"PulseTime8":{"Set":0,"Remaining":0}}
06:41:01.523 MQT: stat/H801_B94A84/RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}
06:41:01.571 MQT: stat/H801_B94A84/RESULT = {"ButtonDebounce":50}
06:41:01.621 MQT: stat/H801_B94A84/RESULT = {"BlinkCount":10}
06:41:02.037 MQT: stat/H801_B94A84/RESULT = {"PulseTime3":{"Set":0,"Remaining":0}}
06:41:02.088 MQT: stat/H801_B94A84/RESULT = {"PulseTime7":{"Set":0,"Remaining":0}}
06:41:02.141 MQT: stat/H801_B94A84/RESULT = {"GPIO0":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"}}
06:41:02.180 MQT: stat/H801_B94A84/RESULT = {"PulseTime1":{"Set":0,"Remaining":0}}
06:41:02.231 MQT: stat/H801_B94A84/RESULT = {"Interlock":"OFF"}
06:41:02.280 MQT: stat/H801_B94A84/RESULT = {"PulseTime5":{"Set":0,"Remaining":0}}
06:41:02.435 QPC: Reset
06:41:04.412 MQT: tele/H801_B94A84/STATE = {"Time":"2023-02-13T06:41:04","Uptime":"0T00:00:10","UptimeSec":10,"Heap":23,"SleepMode":"Dynamic","Sleep":10,"LoadAvg":99,"MqttCount":1,"POWER":"ON","Dimmer":100,"Color":"FFFFFFFF00","HSBColor":"0,0,100","White":100,"CT":153,"Channel":[100,100,100,100,0],"Scheme":0,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"dims","BSSId":"BE:FB:E4:44:09:D2","Channel":6,"Mode":"11n","RSSI":82,"Signal":-59,"LinkCount":1,"Downtime":"0T00:00:04"}}
06:41:08.984 MQT: stat/H801_B94A84/RESULT = {"POWER":"ON"}
06:41:08.989 MQT: stat/H801_B94A84/POWER = ON
06:41:09.039 MQT: stat/H801_B94A84/RESULT = {"POWER":"ON","Dimmer":100,"Color":"FFFFFFFF00","HSBColor":"0,0,100","White":100,"CT":153,"Channel":[100,100,100,100,0]}
06:41:14.441 MQT: stat/H801_B94A84/RESULT = {"POWER":"ON"}
06:41:14.446 MQT: stat/H801_B94A84/POWER = ON
06:41:14.487 MQT: stat/H801_B94A84/RESULT = {"POWER":"ON","Dimmer":100,"Color":"FFFFFFFF00","HSBColor":"0,0,100","White":100,"CT":153,"Channel":[100,100,100,100,0]}

Best regards

**Comment by Jason2866** [comment]
@delinend please open an new issue, you have no connection issues.

**Comment by joba-1** [comment]
@wchulze thanks for shattering our beliefs with facts :+1: , please continue.

Did you also measure the voltage over the esp along with the current spikes?
If it is really a failing power supply it should go down more than usual during the spikes.

**Comment by joba-1** [comment]
And for those who really have a weak power supply:

The most power (by far) is used for sending data. It can help to reduce logging to web console, syslog or mqtt. Use serial for debugging.

**Comment by wchulze** [comment]
@oba-1 Yes, of course I also ocillophed the 3.3 V voltage. There are no break-ins to be seen. Just a some noise, it's not worth taking a picture for.

**Comment by delinend** [comment]
> @delinend please open an new issue, you have no connection issues.

@Jason2866 Thank. I'lll do that. But still think, that it's the same problem, with verry bad WiFi connection, and 12.3.x

Have just tryed to feed a brand new SONOFF Basic R2 and an old SONOFF Basic with an external 3.3V powersupply, whith the LDO dismounted, and same-same. About 25% chance, to connect to the SONOFF via AP mode (6x botton WiFi reset) and a mobil phone. I have to try many times, before I get luke, to connect to the device, from phone.

Btw. A simple test from me, is to PING the devices, on a WiFi quiet WiFi SSID/channel network. It easy to see the differency, on v. 12.2.x and 12.3.x, over a span og 5-10 minutes. PING time comes i cycles, with over 10ms and sometimes Time-out, with 12.3.x With 12.2.x, there is verry little changes, and never over 10ms.

Following devices where I see 12.3.x problem.: H801 (see my first post), Shelly 1v2, SONOFF Basic old, and R2. All devices, as I have no other..

I think that 12.3.x is busy with something, and it's not making coffe. :-)

Best regards

**Comment by wchulze** [comment]
**@delinend** Is the new Sonnoff the V1.3 / R2 and not the ancient type V1.0 R2 ? There is sometime confusion about the versions.

**Comment by delinend** [comment]
@wchulze For the Basic model's, I have tested the very old version, with ESP8266+1Mflash and PCB shunt, and the new R2, with the ESP8285 + blue shunt wire (see photo). See same AP difficulty to connect via mobil phone, and strange PING results, over time.

Btw. I have some ESP-01 and ESP-12 projects, with ESP8266+4M flash, where I now have tryed testing 12.3.x on them, and I'm sorry. I see same behaviour, with 12.3.x

Regarding my home.. It's completly full of H801 moduls and LED strips all over. Ex in one room, I have 6  H801 units, that light up, on entering, via a PIR sensor (MQTT + Domoticz). With 12.3.x there is a painfull delay when the strips is turning on/off. Ex. 2 strips turn on at the same time time, and then 3 more, and then the last strip. This under 1 sec. With 12.2.x all 6 LED strips is turning on/off, with allmost no delay.
And as I descriped, when powering the PWM in high load, all H801 units is doing Hardware_watchdog reboot, in an interval between 2-10 minutes, and the H801 is lightning all LED's on reboot. And I see the LED strips blinking.

The H801 modul, is giving the ESP a heavy load, when running PWM's

Regarding AP's: Have trued testing with Ubiquiti UniFi AP's and an old Asus AP Router. No diff. on PING's.

![image](https://user-images.githubusercontent.com/5739273/218640062-ec98d983-dc28-403c-b35a-4824fec2e38a.png)

**Comment by wchulze** [comment]
To me this looks like a problem with Tasmota version 12.3.1. Hopefully the Tasmota team will find a solution to the problem

**Comment by delinend** [comment]
All my problems with 12.3.x is gone, after upgrade to 12.4.x 😀

Thanks for the god work 👍

**Comment by gfn256** [comment]
All my problems went away by connecting the TH16 directly to the router, not through extender.
Its rather peculiar since I have numerous Tasmota devices connected through the extender, and none had ANY connection problems excepting for the TH16 and also one ESP32-S2 device. Upon connecting these 2 directly to router - All problems gone.
My connection/disconnection problems appeared to be DNS related.

**Comment by Genesisfm** [comment]
I still have a problem with the Sonoff Basic, i'm using the newest firmware:
Tasmota NL 12.4.0
ESP8266EX
The connection is instable, somtimes it works and then the connection is lost:

00:06:36.507 WIF: Verbinding mislukt omdat AP onbereikbaar is
00:06:36.845 WIF: Verbinden met AP1 FearFM_IOT in stand 11n als tasmota-A82BBE-3006...
00:06:46.791 WIF: Verbinding mislukt omdat AP onbereikbaar is
00:06:48.127 WIF: Verbinden met AP1 FearFM_IOT in stand 11n als tasmota-A82BBE-3006...

I tried a lot of things such as rebooting the network (only 2.4ghz).
I'm using Ubiqiuti Unifi UAP-AC-LR AP

**Comment by Genesisfm** [comment]
> I still have a problem with the Sonoff Basic, i'm using the newest firmware: Tasmota NL 12.4.0 ESP8266EX The connection is instable, somtimes it works and then the connection is lost:
> 
> 00:06:36.507 WIF: Verbinding mislukt omdat AP onbereikbaar is 00:06:36.845 WIF: Verbinden met AP1 FearFM_IOT in stand 11n als tasmota-A82BBE-3006... 00:06:46.791 WIF: Verbinding mislukt omdat AP onbereikbaar is 00:06:48.127 WIF: Verbinden met AP1 FearFM_IOT in stand 11n als tasmota-A82BBE-3006...
> 
> I tried a lot of things such as rebooting the network (only 2.4ghz). I'm using Ubiqiuti Unifi UAP-AC-LR AP

Fixed by using the lite version of Tasmota!

**Comment by gfn256** [comment]
@Genesisfm if the lite version works, this would appear to prove its a power issue as suggested above by @Jason2866 https://github.com/arendst/Tasmota/issues/17500#issuecomment-1370927136
